contracts/StablecoinConverter.sol

```diff
@@ -200,15 +202,17 @@ contract StablecoinConverter is EpochTokenLocker {
     function updateRemainingOrder(
         address owner,
         uint orderId,
-        uint128 executedSellAmount
+        uint128 volume
     ) internal returns (uint) {
-        Order memory order = orders[owner][orderId];
-        uint128 newSellAmount = uint128(order.sellAmount.sub(executedSellAmount));
-        orders[owner][orderId].buyAmount = uint128(
```

Comment:
The fundamental problem here, is that we keep the price ratio of the original limit order, even if a partial fulfilment achieved a much better price.

I'd argue it's fine to get your limit price averaged on the full order (after all you specified this limit for your full order, so you should be happy with it).

Reply:
You think if we get a good price in the first trade, we should recalculate the price to a worse price that if we full fill all the remainings at that price, we still fulfill the original?

Yep, I think if it's not too complicated/expensive to update the price it makes sense. 
Maybe it would be strange for the user on how to present this information, cause if the user see just the "standing order" with his remaining volume and price, he's might be confused, but we could come up on how to show that...

Reply:
I think a good argument by @koeppelmann was that in the stableX use case, you might have a large standing order at $1.01 and only because half of it was fulfilled at $1.10, you don't want the second half to be fulfilled at $0.92